### PR TITLE
refactor(dispatch,dispatch-service): added post-processing to stdout log

### LINKED ISSUE #3451: Simplify tracking of post-processing failures
Currently, post-processing failures can only be distinguished from the status reason. It would be helpful to create a distinct status code for this failure.

- [ ] Create distinct status code
- [ ] Update `apps/dispatch/src/app/components/simulations/view/simulation-log/simulation-log.component.ts` to use this code to format post-processing errors

### PR DESCRIPTION
**Pull request by jonrkarr**
- [x] Organized console formatting commands
- [x] Cleaned up log displaying
- [x] Added post-processing status to combined stdout/err log -- closes #3451 
- [x] Enabled log PUT method
- [x] Improve handling of case when YAML log file doesn't exist -- treat this as an optional simulator feature rather than an error
- [x] Handle errors in logs by posting stdout
- [x] Improve log messages
- [x] Encapsulate arguments in SBATCH script in quotes

### CODE REVIEW

File: apps/dispatch-service/src/app/submission/complete.processor.ts [hunk 1]
```diff
@@ -152,21 +153,49 @@ export class CompleteProcessor {
...
+
+    const extraStdLog = (
+      ''
+      + '\n'
+      + `\n${ConsoleFormatting.cyan}======================= Post-processing simulation run ======================${ConsoleFormatting.noColor}`
+      + `\n${statusColor}statusReason${statusEndColor}`
+      + '\n'
+      + `\n${ConsoleFormatting.cyan}============ Run complete. Thank you for using runBioSimulations! ===========${ConsoleFormatting.noColor}`
+    );
+    this.logService.createLog(id, extraStdLog, true)
```
Comment (bilalshaikh42) [inline apps/dispatch-service/src/app/submission/complete.processor.ts]: This does not seem right. I think this was already an issue before that just got copied over, but the then does not catch errors, but successful updates

**Comment by bilalshaikh42** [comment]
What is the intention of the put method? Is this being called by the dispatch service?

**Review by bilalshaikh42 — APPROVED** [review]
LGTM

File: apps/dispatch-service/src/app/submission/complete.processor.ts (hunk 1 above)
Comment (jonrkarr) [reply apps/dispatch-service/src/app/submission/complete.processor.ts]: Errors are handled by `const logPostSucceeded = logProcessingResult.status === 'fulfilled' && logProcessingResult?.value;` and the second call to `createLog` (`this.logService.createLog(id, !logPostSucceeded, extraStdLog, true)`)

**Pull request merged**